components/support/nimbus-fml/src/defaults/validator.rs

```diff
@@ -222,6 +244,254 @@ impl<'a> DefaultsValidator<'a> {
             }),
         }
     }
+
+    #[allow(clippy::too_many_arguments)]
+    fn validate_string_aliases(
+        &self,
+        path: &str,
+        error_path: &[String],
+        typ: &TypeRef,
+        value: &Value,
+        defn: &HashMap<&str, &PropDef>,
+        skip: &Option<TypeRef>,
+        errors: &mut Vec<FMLError>,
+    ) {
+        // As an optimization (to stop validating the definition against itself),
+        // we want to skip validation on the `skip` type ref: this is only set by the property defining
+        // a string-alias.
+        let should_validate = |v: &TypeRef| -> bool { !matches!(skip, Some(t) if t == v) };
+        match (typ, value) {
+            (TypeRef::StringAlias(_), Value::String(s)) => {
+                check_string_aliased_property(path, error_path, typ, s, defn, errors)
+            }
+            (TypeRef::Option(_), &Value::Null) => (),
+            (TypeRef::Option(inner), _) => {
+                self.validate_string_aliases(path, error_path, inner, value, defn, skip, errors)
+            }
+            (TypeRef::List(inner), Value::Array(array)) => {
+                if should_validate(inner) {
+                    for value in array {
+                        self.validate_string_aliases(
+                            path, error_path, inner, value, defn, skip, errors,
+                        );
+                    }
+                }
+            }
+            (TypeRef::EnumMap(kt, vt), Value::Object(map)) => {
+                if should_validate(kt) && matches!(**kt, TypeRef::StringAlias(_)) {
+                    for value in map.keys() {
+                        check_string_aliased_property(path, error_path, kt, value, defn, errors);
+                    }
+                }
+
+                if should_validate(vt) {
+                    for (key, value) in map {
+                        let path = format!("{path}['{key}']");
+                        let error_path = append_quoted(error_path, key);
```

Comment:
We do things like this a *lot*. It might be nice in the future to have a Path type as a wrapper around this everywhere, whith methods for recording path traversal through the FML. It could also mean we wouldn't have to pass `path` and `error_path` to `validate_*` and `check_*` functions as it could take on the responsibility of both of those parameters.

Not for this PR, just a thought for follow-ups.

Reply:
Oh that's a good idea. Agree, not for this PR however.